packages/lodestar/src/network/reqresp/rateTracker.ts

```diff
@@ -0,0 +1,47 @@
+import {MapDef} from "../../util/map";
+
+/**
+ * The generic rate tracker allows up to `limit` objects in a period of time.
+ * This could apply to both request count or block count, for both requests and responses.
+ */
+export class RateTracker {
+  private requestsWithinWindow = 0;
+  private requests = new MapDef<number, number>(() => 0);
```

Comment:
What do each number in `<number, number>` represent? Add JSDocs to `requests` explaining